### DISCUSSION TITLE
Why are some environmental conditions arbitrarily discretised?

### DISCUSSION THREAD

**Discussion by mansoor2016** [OpenSimulationInterface/open-simulation-interface#738] [General]
The precipitation intensity is provided as a discrete enum. why? It is not consistent for all env. conditions. 

If you develop regression models for environmental effects, likely you will need continuous inputs. 

Providing a double value field + a confidence level field would be a more useable design.

**Comment by jdsika** [comment]
Hey! 
If you look at the PRs you can find some discussions around the topic and also one with your suggestion: https://github.com/OpenSimulationInterface/open-simulation-interface/pull/582/files

Environmental conditions are often not measured very precisely, imagine a real world drive somewhere and the driver would need to document and classify the current rain intensity. Having a double field would just create the illusion of precision. If your modlling requires discrete values I recommend to make a proposal to add e.g additional continous values without breaking backwards compatibility and allowing for more precise values not allowing them to contradict. 

> new_value = 0.3
> precipitation = PRECIPITATION_VERY_LIGHT 
> 
> // Very light intensity of precipitation, when I in [0.1,0.5[ mm/h
> //
> PRECIPITATION_VERY_LIGHT = 3;

There is also a modelling working group and I suppose they have formed some opinions around it alrady. @PhRosenberger may know something here?

Best regards
Carlo

**Comment by PhRosenberger** [comment]
Yes, I have already collected several discussions/issues/PRs on that topic and I would be glad if you, @mansoor2016, would join our project to elaborate on that topic!

Here is a starting point for that: https://github.com/OpenSimulationInterface/open-simulation-interface/pull/601

**Comment by mansoor2016** [comment]
Looks like the suggestions in #601 would resolve this discussion, closing this now. 

Thanks @jdsika & @PhRosenberger
